libbeat/outputs/elasticsearch/client.go

```diff
@@ -104,9 +110,10 @@ func (client *Client) PublishEvents(
 	}
 
 	// check response for transient errors
-	events = bulkCollectPublishFails(res, events)
-	if len(events) > 0 {
-		return events, mode.ErrTempBulkFailure
+	failed_events := bulkCollectPublishFails(res, events)
+	ackedEvents.Add(int64(len(events) - len(failed_events)))
+	if len(failed_events) > 0 {
+		return failed_events, mode.ErrTempBulkFailure
```

Comment:
some additional metrics might be of interest:
- number of PublishEvents calls
- number of nonACKed events